### PR TITLE
React 16

### PR DESCRIPTION
**Pull request by dwhitney**
This upgrades Thermite to the latest versions of all of its dependencies along with React 16.02. 

The latest `purescript-react` added a `render` type variable to its `Render` type, so I fixed `ReactElement` to it, which I believe will work, but someone should review. 

Thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
+    "create-react-class": "^15.6.1",
+    "react": "^16.0.2",
+    "react-dom": "^16.0.2"
   },
   "devDependencies": {
-    "pulp": "^11.0.0",
-    "purescript-psa": "^0.5.0",
-    "purescript": "^0.11.1",
-    "rimraf": "^2.5.4"
+    "bower": "^1.8.2",
```
Comment (safareli) [inline package.json]: Why add bower?

**Comment by dwhitney** [comment]
it's gone now

**Comment by dwhitney** [comment]
I was working with thermite a bit more today and realized that `createReactSpec` adds a redundant div, which is kind of annoying. This can be gotten rid of if the `ReactRender` type is fed an `Array React.ReactElement`. I think this would be better, but probably needs discussion

**Comment by athanclark** [comment]
@dwhitney I've uploaded thermite v6.0.1 which is purs v0.12.x compatible, and react v16.x compatible. Sorry for this wasted effort!

**Pull request closed without merging**